codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -1175,6 +1209,32 @@ public ChannelConfig setRecvByteBufAllocator(RecvByteBufAllocator allocator) {
             super.setRecvByteBufAllocator(allocator);
             return this;
         }
+
+        @Override
+        public Map<ChannelOption<?>, Object> getOptions() {
+            return getOptions(
+                    super.getOptions(),
+                    Http2StreamChannelOption.AUTO_WRITE_WINDOW_UPDATE_FRAME);
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> T getOption(ChannelOption<T> option) {
+            if (option == Http2StreamChannelOption.AUTO_WRITE_WINDOW_UPDATE_FRAME) {
+                return (T) Boolean.valueOf(autoFlowControl);
+            }
+            return super.getOption(option);
+        }
+
+        @Override
+        public <T> boolean setOption(ChannelOption<T> option, T value) {
+            validate(option, value);
+            if (option == Http2StreamChannelOption.AUTO_WRITE_WINDOW_UPDATE_FRAME) {
+                autoFlowControl = (Boolean) value;
```

Comment:
In my minds eye, `autoFlowControl` is somewhat analogous to `autoRead`: when we switch from `autoFlowControl == false` to true, do we need to potentially send a window update? _If_ we do, it could well be a follow up.

Reply:
it will do so automatically in this case as part of the reading.

Reply:
Is that true if the flow window has been completely depleted and we don't receive any more frames until we send a window update?

Reply:
ha... actually it didn't. Fixed